src/player.cpp

```diff
@@ -33,22 +34,101 @@ void Player::setStateChangedCallback(void (*stateChangedCallback)(unsigned int))
     soloud.setStateChangedCallback(stateChangedCallback);
 }
 
-PlayerErrors Player::init(unsigned int sampleRate, unsigned int bufferSize, unsigned int channels)
+PlayerErrors Player::init(unsigned int sampleRate, unsigned int bufferSize, unsigned int channels, int deviceID)
 {
     if (mInited)
         return playerAlreadyInited;
 
+    void *playbackInfos_id = nullptr;
+    if (deviceID != -1)
+    {
+        // Calling this will init [pPlaybackInfos]
+        auto const devices = listPlaybackDevices();
+        if (devices.size() == 0 || deviceID >= devices.size())
+            return backendNotInited;
+        playbackInfos_id = &pPlaybackInfos[deviceID].id;
+    }
+
     // initialize SoLoud.
     SoLoud::result result = soloud.init(
         SoLoud::Soloud::CLIP_ROUNDOFF,
-        SoLoud::Soloud::MINIAUDIO, sampleRate, bufferSize, channels);
-    if (result == SoLoud::SO_NO_ERROR)
+        SoLoud::Soloud::MINIAUDIO, sampleRate, bufferSize, channels, playbackInfos_id);
+    if (result == SoLoud::SO_NO_ERROR){
         mInited = true;
-    else
+        mSampleRate = sampleRate;
+        mBufferSize = bufferSize;
+        mChannels = channels;
+    } else
         result = backendNotInited;
     return (PlayerErrors)result;
 }
 
+PlayerErrors Player::changeDevice(int deviceID)
+{
+    if (!mInited)
+        return backendNotInited;
+
+    void *playbackInfos_id = nullptr;
+
+    // Calling this will init [pPlaybackInfos]
+    auto const devices = listPlaybackDevices();
+    if (devices.size() == 0 || deviceID >= devices.size())
+        return backendNotInited;
+    playbackInfos_id = &pPlaybackInfos[deviceID].id;
+
+    SoLoud::result result = soloud.miniaudio_changeDevice(playbackInfos_id);
+    
+    if (result != SoLoud::SO_NO_ERROR)
+        result = backendNotInited;
```

Comment:
A more descriptive error here, too. Something like `playbackDeviceChangeError`? Or whatever the SoLoud::result can be in this scenario.

Reply:
I looked at the code and seems the only error could be `UNKNOWN_ERROR`.

Reply:
Fair enough, thanks for checking!

Reply:
I'd still prefer a more meaningful error or at least a comment here in the code ("miniaudio_changeDevice can only throw UNKNOWN_ERROR"). But that's a nit.

Reply:
added a comment